### PR TITLE
Image download headings

### PR DESCRIPTION
**Pull request by pandu-k**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Feature

* **What is the current behavior?** (You can also link to an open issue here)
No way to pass through image download auth information during add_docs and search

* **What is the new behavior (if this is a feature change)?**
Pass through image download auth information during add_docs and search

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Have unit tests been run against this PR?** (Has there also been any additional testing?)
In progress

* **Related Python client changes** (link commit/PR here)
https://github.com/marqo-ai/py-marqo/pull/60

* **Related documentation changes** (link commit/PR here)
to do 

* **Other information**:

* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: src/marqo/tensor_search/api.py [hunk 1]
```diff
@@ -131,14 +131,16 @@ def add_or_replace_documents(docs: List[Dict], index_name: str, refresh: bool =
                         marqo_config: config.Config = Depends(generate_config),
                         batch_size: int = 0, processes: int = 1,
                         non_tensor_fields: List[str] = Query(default=[]),
-                        device: str = Depends(api_validation.validate_device)):
+                        device: str = Depends(api_validation.validate_device),
+                        image_download_headers: str = "{}"):
```
Comment (pandu-k) [inline src/marqo/tensor_search/api.py]: This should be a dict. We should let fastAPI parse it

**Comment by pandu-k** [comment]
Unit tests: https://github.com/marqo-ai/marqo/actions/runs/4221002340

**Comment by pandu-k** [comment]
Manually tested (cURL + py-marqo PR https://github.com/marqo-ai/py-marqo/pull/60) : 
- add_documents with auth-required images
- searching for docs with auth-required image

File: src/marqo/tensor_search/parallel.py [hunk 2]
```diff
@@ -165,7 +171,8 @@ def get_threads_per_process(processes: int):
 
 def add_documents_mp(config=None, index_name=None, docs=None, 
                      auto_refresh=None, batch_size=50, processes=1, device=None,
-                     non_tensor_fields: List[str] = [], update_mode: str = None):
+                     non_tensor_fields: List[str] = [], update_mode: str = None,
+                     image_download_headers: str = "{}", use_existing_vectors=None):
```
Comment (tomhamer) [inline src/marqo/tensor_search/parallel.py]: still using string here

**Comment by pandu-k** [comment]
Tests running after merging mainline back in:
- https://github.com/marqo-ai/marqo/actions/runs/4221885749
- https://github.com/marqo-ai/marqo/actions/runs/4221887784

**Review by tomhamer — APPROVED** [review]

**Pull request merged**